### PR TITLE
Feature/favorites page

### PR DESCRIPTION
**Pull request by MavOmidK**
Works, tell me if theres anything I missed or bugs.

### CODE REVIEW

File: src/providers/add-movies.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { AngularFire, FirebaseListObservable } from 'angularfire2';
+
+@Injectable()
+
+export class AddMovies {
+  public favoritesList: FirebaseListObservable<any>;
+  public userId: string;
+  public favorites: Array<any>;
+
+  constructor(public af: AngularFire) {
```
Comment (aagavin) [inline src/providers/add-movies.ts]: I'm not sure what this file does??

**Review by aagavin — CHANGES_REQUESTED** [review]
Looks good. Can you fix merge conflicts.

**Review by antonybogun — CHANGES_REQUESTED** [review]
1. TSLint gives me these errors:
`[12:24:09]  tslint: ...ect/COMP308-Group2-Favourite-Movie-List/src/pages/movie-details/movie-details.ts, line: 6
[12:24:09]  tslint: ...08-W2017-Project/COMP308-Group2-Favourite-Movie-List/src/providers/add-movies.ts, line: 1
[12:24:09]  tslint: ...08-W2017-Project/COMP308-Group2-Favourite-Movie-List/src/providers/add-movies.ts, line: 2
[12:24:09]  tslint: ...08-W2017-Project/COMP308-Group2-Favourite-Movie-List/src/providers/add-movies.ts, line: 2
            Unused import: 'FirebaseObjectObservable'

       L5:  // import angularfire
       L6:  import { AngularFire, FirebaseListObservable, FirebaseObjectObservable } from 'angularfire2';

            Unused import: 'Component'

       L1:  import { Component } from '@angular/core';
       L2:  import { NavController, NavParams } from 'ionic-angular';

            Unused import: 'NavController'

       L1:  import { Component } from '@angular/core';
       L2:  import { NavController, NavParams } from 'ionic-angular';
       L3:  import {Injectable} from "@angular/core";

            Unused import: 'NavParams'

       L1:  import { Component } from '@angular/core';
       L2:  import { NavController, NavParams } from 'ionic-angular';
       L3:  import {Injectable} from "@angular/core";

[12:24:09]  lint finished in 5.40 s`

2. Can you please update 'Add to favorites' button on 'Movie Details' page to 'Remove from favorites' once clicked. It will give user a chance to make sure that movie was added successfully without going to 'Favorites' page

**Comment by eddieysong** [comment]
It looks like Omid missed a commit in `dev`, so I tried to resolve the merge conflicts by including new lines from both branches. There doesn't seem to be any deletions, so it should work... hopefully it doesn't mess up other branches..

**Review by eddieysong — APPROVED** [review]
Tested everything in `ionic serve`, functionalities seem to be complete and they play nicely. One exception was thrown by AuthData (auth-data.ts), but I cannot recreate it so it might have just been a firebase thing.

**Review by antonybogun — APPROVED** [review]

**Review by aagavin — APPROVED** [review]

**Review by aagavin — APPROVED** [review]

**Pull request merged**